src/Serialization/upgrades/0.11.3.jl

```diff
@@ -0,0 +1,118 @@
+################################################################################
+# Upgrade Summary
+# When `is_basic_serialization_type(T) == true` the type `Vector{T}` the entries
+# are no longer serialized as dicts, The type is encoded on the Vector at the same level as
+# the entry data. The entries no longer being dictionaries prevents the use of backrefs.
+
+
+push!(upgrade_scripts, UpgradeScript(
+    v"0.11.3", # version this script upgrades to
+    function upgrade_0_11_3(s::DeserializerState, dict::Dict)
+        # file comes from polymake
+        haskey(dict, :_ns) && haskey(dict[:_ns], :polymake) && return dict
+
+        # moves down tree to point where type exists in dict
+        # since we are only doing updates based on certain types
+        # no :type key implies the dict is data
+        if !haskey(dict, :type)
+            upgraded_dict = Dict{Symbol, Any}()
+            for (key, value) in dict
+                if value isa String
+                    upgraded_dict[key] = value
+                elseif value isa Dict{Symbol, Any}
+                    upgraded_dict[key] = upgrade_0_11_3(s, value)
+                else
+                    values = []
+                    for v in value
+                        if v isa String
+                            push!(values, v)
+                        else
+                            push!(values, upgrade_0_11_3(s, v))
+                        end
+                    end
+                    upgraded_dict[key] = values
+                end
+            end
+            return upgraded_dict
+        end
+
+        if dict[:type] == string(backref_sym)
+            backref = s.objs[UUID(dict[:id])]
+
+            # data no longer uses backref and is part of the update
+            backref isa String && return backref
+
+            return dict
+        end
+
+        # this handles types like FlintRationalField that have no associated data
+        !haskey(dict, :data) && return dict
+
+        # apply upgrade to vector entries
+        # we only do this here since upgrade_0_11_13 is only defined for dicts
+        # we could make this it's own function outside of this script
+        if dict[:type] == "Vector"
+            upgraded_vector = []
+            entry_type = nothing
+
+            for entry in dict[:data][:vector]
+                result = upgrade_0_11_3(s, entry)
+
+                # store values in state that are vector entries that aren't backrefs
+                if entry[:type] != string(backref_sym)
+                    if haskey(entry, :id)
+                        s.objs[UUID(entry[:id])] = result
+                    end
+                    entry_type = entry[:type]
```

Comment:
Maybe insert a paranoia check?
```suggestion
                    @assert entry_type === nothing || entry_type == entry[:type]
                    entry_type = entry[:type]
```